### DISCUSSION TITLE
Specified remote tier is already in use - issue on new storage using old opensearch and S3 bucket

### DISCUSSION THREAD

**Discussion by robert-pudlowski-mox** [minio/minio#18011] [Q&A] [answered]
Issue on this scenario: 
1. Minio is working on EKS (PVC storage), remote tier is configured with S3 bucket, opensearch is configured and fully working
2. I'm recreating PVC - storage where OpenCTI bucket is located.
3. Executing the command to check if the remote tier is still exist: mc ilm tier myminio -> the result is that there is no remote tier what is expected since the PVC was recreated.
4. Trying to add the remote tier the same as it was before recreating PVC
5. Getting error after executing this command: mc ilm tier add s3 myminio S3 .... : **mc: <ERROR> Unable to configure remote tier target. Specified remote tier is already in use.**

**I don't want to USE data on minio as it was before. I just want to add remote tier like for a new cluster. Does it mean that every time when PVC is recreated, I need to create new OpenSearch domain or S3 bucket? I can not reuse the old one? I'm not afraid of loosing my data. Sometimes the PVC can be crashed on EKS it need to be recreated, but it should be able to configure remote tier from scratch using old OpenSearch and S3 bucket. Now, I can't.** 

I'm not able to add remote tier since mc is showing me that remote tier is not existing but when I'm trying to create the remote tier it is getting me the error that it is already in use. It looks like there are some leftover of the configuration and mc client is not able to manage with it. Maybe something need to be removed from OpenSearch? 

## Expected Behavior
I would like to add the same Remote Tier for ILM after PVC is recreated. 

## Current Behavior
I'm not able to add remote tier since mc is showing me that remote tier is not existing but when I'm trying to create the remote tier it is getting me the error that it is already in use. It looks like there are some leftover of the configuration and mc client is not able to manage with it. Maybe something need to be removed from OpenSearch? 

## Your Environment
* Version used (`minio --version`): RELEASE.2023-05-18T00-05-36Z
* Server setup and configuration: EKS AWS using PVC for persistent storage

**Comment by robert-pudlowski-mox** [comment]
I don't know why, but minio is thinking that the S3 remote tier is used by another tenant - but it is not.

**Comment by robert-pudlowski-mox** [comment]
Please don't close this issue, I want to ask some questions here.

**Comment by harshavardhana** [comment]
> I don't know why, but minio is thinking that the S3 remote tier is used by another tenant - but it is not.

Please don't re-open issues that should be closed. Moving this to a discussion to provide more clarification.

**Comment by harshavardhana** [comment, marked as answer]
> **I don't want to USE data on minio as it was before. I just want to add remote tier like for a new cluster. Does it mean that every time when PVC is recreated, I need to create new OpenSearch domain or S3 bucket? I can not reuse the old one? I'm not afraid of loosing my data. Sometimes the PVC can be crashed on EKS it need to be recreated, but it should be able to configure remote tier from scratch using old OpenSearch and S3 bucket. Now, I can't.**

No you cannot re-use the s3-tiered backend again in a fresh PVC. Because you deleted all the metadata that was stored on the PVC.

**Comment by IGZangelcostales** [comment]
I make a metadata backup for all my buckets with: 
`mc admin cluster bucket export myminio/`
I get this:
`mc: Bucket metadata successfully downloaded as myminio-bucket-metadata.zip`

then I deleted the minio and the volume, reinstalled.. and started with a fresh minio. now I tried to restore the buckets and objetcs (that still exist in the external cloud storage).. by restoring the metadata with:
`mc admin cluster bucket import myminio/ ./myminio-bucket-metadata.zip`

I get this:
`Name: bucket1
  Object lock:  ✔
  Versioning:  ✔
  Lifecycle:  ✗
  Notification:  ✔
  Quota:  ✔`

and as you can see, all was restored, except the Lifecycle because it depends on the tier... :( 
so I tried to recreate the tier, but... `Specified remote tier is already in use`

¿how can I backup/restore the tier config?
Thanks!

**Reply by IGZangelcostales** [reply]
I answer myself:
   add the tier again with `--force` parameter.. xD
